### PR TITLE
Number of sequences over time: implement line chart view

### LINKED ISSUE #317: Number sequences over time: line chart

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #317

### Summary

Adding line charts.

### Screenshot

image

image

image

### PR Checklist

- [x] All necessary documentation has been adapted.
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

**Comment by chaoran-chen** [comment]
I think that filling up with 0 makes sense, is there a reason why you wouldn't fill it up?

**Comment by fengelniederhammer** [comment]
> I think that filling up with 0 makes sense, is there a reason why you wouldn't fill it up?

To indicate that there is not data (as opposed to data is 0). But it doesn't matter much. For the bar chart, I had to fill everything with 0s anyway for technical reasons.

**Comment by chaoran-chen** [comment]
No data in this case means 0 available sequences, so I think that we should fill up with 0.

File: components/src/preact/numberSequencesOverTime/number-sequences-over-time-line-chart.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const getDatasets = (data: NumberOfSequencesDatasets) => {
+    const tableData = getNumberOfSequencesOverTimeTableData(data, 'date');
+
+    return data.map(
+        ({ displayName }, index) =>
+            ({
+                borderWidth: 1,
+                pointRadius: 0,
+                label: displayName,
+                backgroundColor: singleGraphColorRGBAById(index, 0.3),
```
Comment (JonasKellerer) [inline components/src/preact/numberSequencesOverTime/number-sequences-over-time-line-chart.tsx:51]: Do we need the backgroundColor when we dont have a confidence interval (yet)?

File: components/tests/visualizationStories.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export const visualizationStories = [
     { id: 'visualization-mutation-comparison--venn-diagram', title: 'Mutation comparison' },
     { id: 'visualization-aggregate--table', title: 'Aggregate', testDownloadWithFilename: 'aggregate.csv' },
     { id: 'visualization-number-sequences-over-time--one-dataset-bar-chart', title: 'Number of sequences of time' },
+    { id: 'visualization-number-sequences-over-time--one-dataset-line-chart', title: 'Number of sequences of time' },
```
Comment (JonasKellerer) [inline components/tests/visualizationStories.ts:34]:
```suggestion
    { id: 'visualization-number-sequences-over-time--one-dataset-line-chart', title: 'Number of sequences over time' },
```

**Review by JonasKellerer — APPROVED** [review]
LGTM

File: components/src/preact/numberSequencesOverTime/number-sequences-over-time-line-chart.tsx (hunk 1 above)
Comment (fengelniederhammer) [reply components/src/preact/numberSequencesOverTime/number-sequences-over-time-line-chart.tsx:51]: Actually yes. It shows up in the little colored box in the tooltip.

**Pull request merged**